### PR TITLE
[NFC] Use references to avoid copying

### PR DESCRIPTION
**Pull request by pratyay-p**
Modifying `auto` to `auto&` to avoid unnecessary copying

### CODE REVIEW

**Comment by pratyay-p** [comment]
Adding @phoebewang , @KanRobert , @MalaySanghi

**Comment by pratyay-p** [comment]
Ping

**Review by MalaySanghi — APPROVED** [review]

File: llvm/lib/CodeGen/RegAllocFast.cpp [hunk 1]
```diff
@@ -584,7 +584,7 @@ void RegAllocFastImpl::spill(MachineBasicBlock::iterator Before,
       SpilledOperandsMap;
   for (MachineOperand *MO : LRIDbgOperands)
     SpilledOperandsMap[MO->getParent()].push_back(MO);
-  for (auto MISpilledOperands : SpilledOperandsMap) {
+  for (auto& MISpilledOperands : SpilledOperandsMap) {
```
Comment (topperc) [inline llvm/lib/CodeGen/RegAllocFast.cpp:587]: LLVM style should be `auto &MISpilledOperands`

Comment (pratyay-p) [reply llvm/lib/CodeGen/RegAllocFast.cpp]: Done, fixed formatting in both files

**Review by phoebewang — APPROVED** [review]
LGTM.

File: llvm/lib/CodeGen/RegAllocFast.cpp [hunk 2]
```diff
@@ -584,7 +584,7 @@ void RegAllocFastImpl::spill(MachineBasicBlock::iterator Before,
       SpilledOperandsMap;
   for (MachineOperand *MO : LRIDbgOperands)
     SpilledOperandsMap[MO->getParent()].push_back(MO);
-  for (auto MISpilledOperands : SpilledOperandsMap) {
+  for (auto &MISpilledOperands : SpilledOperandsMap) {
```
Comment (HaohaiWen) [inline llvm/lib/CodeGen/RegAllocFast.cpp:587]: Why not add const here.

Comment (pratyay-p) [reply llvm/lib/CodeGen/RegAllocFast.cpp]: @HaohaiWen Done. Modified to use const

**Pull request merged**